src/kv/test/kv_serialisation.cpp

```diff
@@ -13,70 +14,42 @@
 #include <string>
 #include <vector>
 
-struct CustomClass
+struct RawMapTypes
 {
-  int m_i;
-
-  CustomClass() : CustomClass(-1) {}
-  CustomClass(int i) : m_i(i) {}
-
-  int get() const
-  {
-    return m_i;
-  }
-  void set(std::string val)
-  {
-    m_i = std::stoi(val);
-  }
-
-  CustomClass operator()()
-  {
-    CustomClass ret;
-    return ret;
-  }
-
-  bool operator<(const CustomClass& other) const
-  {
-    return m_i < other.m_i;
-  }
-
-  bool operator==(const CustomClass& other) const
-  {
-    return !(other < *this) && !(*this < other);
-  }
-
-  MSGPACK_DEFINE(m_i);
+  using StringString = kv::Map<std::string, std::string>;
+  using NumNum = kv::Map<size_t, size_t>;
+  using NumString = kv::Map<size_t, std::string>;
+  using StringNum = kv::Map<std::string, size_t>;
 };
 
-namespace std
+struct ExperimentalMapTypes
 {
-  template <>
-  struct hash<CustomClass>
-  {
-    std::size_t operator()(const CustomClass& inst) const
-    {
-      return inst.get();
-    }
-  };
-}
-
-DECLARE_JSON_TYPE(CustomClass)
-DECLARE_JSON_REQUIRED_FIELDS(CustomClass, m_i)
+  using StringString = kv::experimental::Map<std::string, std::string>;
+  using NumNum = kv::experimental::Map<size_t, size_t>;
+  using NumString = kv::experimental::Map<size_t, std::string>;
+  using StringNum = kv::experimental::Map<std::string, size_t>;
+};
 
-TEST_CASE(
+TEST_CASE_TEMPLATE(
```

Comment:
Most of the test cases are now templated, so we test both `kv::Map` and `kv::experimental::Map` through the exact same code. When we rename (and perhaps `kv::Map` becomes untemplated), this will go away.